modules/core/src/main/java/org/apache/ignite/internal/commandline/TracingConfigurationCommand.java

```diff
@@ -0,0 +1,345 @@
+                }
+                case LABEL: {
+                    lb = argIter.nextArg(
+                        "The label should be specified.");
+
+                    break;
+                }
+                case SAMPLING_RATE: {
+                    strVal = argIter.nextArg(
+                        "The sampling rate should be specified. Decimal value between 0 and 1 should be used.");
+
+                    try {
+                        samplingRate = Double.parseDouble(strVal);
+                    }
+                    catch (IllegalArgumentException e) {
+                        throw new IllegalArgumentException(
+                            "Invalid sampling-rate '" + strVal + "'. Decimal value between 0 and 1 should be used.");
+                    }
+
+                    if (samplingRate < SAMPLING_RATE_NEVER || samplingRate > SAMPLING_RATE_ALWAYS)
+                        throw new IllegalArgumentException(
+                            "Invalid sampling-rate '" + strVal + "'. Decimal value between 0 and 1 should be used.");
+
+                    break;
+                }
+                case INCLUDED_SCOPES: {
+                    Set<String> setStrVals = argIter.nextStringSet(
+                        "At least one supported scope should be specified.");
+
+                    for (String scopeStrVal : setStrVals) {
+                        try {
+                            supportedScopes.add(Scope.valueOf(scopeStrVal));
+                        }
+                        catch (IllegalArgumentException e) {
+                            throw new IllegalArgumentException(
+                                "Invalid supported scope '" + scopeStrVal + "'. The following " +
+                                    "values can be used: " + Arrays.toString(Scope.values()) + '.');
+                        }
+                    }
+
+                    break;
+                }
+            }
+        }
+
+        // Scope is a mandatory attribute for all sub-commands except get_all and reset_all.
+        if ((cmd != GET_ALL && cmd != RESET_ALL) && scope == null) {
+            throw new IllegalArgumentException(
+                "Scope attribute is missing. Following values can be used: " + Arrays.toString(Scope.values()) + '.');
+        }
+
+        switch (cmd) {
+            case GET_ALL:
+            case RESET_ALL: {
+                tracingConfigurationArgs.withScope(scope);
+
+                break;
+            }
+
+            case RESET:
+            case GET: {
+                tracingConfigurationArgs.withScope(scope).withLabel(lb);
+
+                break;
+            }
+
+            case SET: {
+                tracingConfigurationArgs.withScope(scope).withLabel(lb).withSamplingRate(samplingRate).
+                    withSupportedScopes(supportedScopes);
+
+                break;
+            }
+
+            default: {
+                // We should never get here.
+                assert false;
```

Comment:
Please add a meaningful message to this assert, something like as follows: `Unexpected tracing configuration argument [arg=` + cmd +]`

Reply:
Fixed.